### PR TITLE
Splunk search

### PR DESCRIPTION
**Pull request by meirwah**
fixes https://github.com/demisto/server/issues/8980
fixes https://github.com/demisto/server/issues/6222

### CODE REVIEW

File: Integrations/integration-SplunkPy.yml [hunk 1]
```diff
@@ -187,5 +212,25 @@ script:
...
+    arguments:
+    - name: query
+      required: true
+      default: true
+      description: The splunk search language string to execute, example :"* | head
+        3"
+    - name: earliest_time
+      description: Specifies the earliest time in the time range to search. The time
+        string can be a UTC time (with fractional seconds), a relative time specifier
+        (to now), or a formatted time string. Default is 1 week ago
```
Comment (asafshen) [inline Integrations/integration-SplunkPy.yml]: I think example should be more clear (e.g. "2 days ago") 
also - tried to give `earliest_time="1 week ago"` and it does not work

File: Integrations/integration-SplunkPy.yml [hunk 2]
```diff
@@ -187,5 +212,25 @@ script:
...
+    - name: latest_time
+      description: Specifies the latest time in the time range to search. The time
+        string can be a UTC time (with fractional seconds), a relative time specifier
+        (to now), or a formatted time string.
```
Comment (asafshen) [inline Integrations/integration-SplunkPy.yml]: I think example should be more clear (e.g. "to now"/ "now")

File: Integrations/integration-SplunkPy.yml [hunk 3]
```diff
@@ -138,6 +138,31 @@ script:
...
+        kwargs_oneshot = {"earliest_time":time}
+        if demisto.get(demisto.args(), 'earliest_time'):
+            kwargs_oneshot['earliest_time'] = demisto.args()['earliest_time']
+        if demisto.get(demisto.args(), 'latest_time'):
+            kwargs_oneshot['latest_time'] = demisto.args()['latest_time']
+        if demisto.get(demisto.args(), 'event_limit'):
+            kwargs_oneshot['count'] = int(demisto.args()['event_limit'])
+        searchquery_oneshot = demisto.args()['query']
+        if not searchquery_oneshot.startswith('search'):
+            searchquery_oneshot = 'search '+searchquery_oneshot
```
Comment (asafshen) [inline Integrations/integration-SplunkPy.yml]: nice

File: Integrations/integration-SplunkPy.yml [hunk 4]
```diff
@@ -138,6 +138,31 @@ script:
...
+
+        reader = results.ResultsReader(oneshotsearch_results)
+        res = []
+        for item in reader:
+            if isinstance(item, results.Message):
+                demisto.results({"Type": 1, "ContentsFormat": "json", "Contents": json.dumps(result.message)})
+            elif isinstance(item, dict):
+                # Normal events are returned as dicts
+                res.append(item)
+        demisto.results({"Type": 1, "ContentsFormat": "json", "Contents": json.dumps(res)})
```
Comment (asafshen) [inline Integrations/integration-SplunkPy.yml]: 1. I think we should present it in markdown table if possible. here is the function I used in other python integration
```
def create_entry(header, table, context = {}, headers = None):
    return {
        'ContentsFormat': formats['json'],
        'Type': entryTypes['note'],
        'Contents': table,
        'ReadableContentsFormat': formats['markdown'],
        'HumanReadable': tableToMarkdown(header, table, headers) if table else 'No result were found',
        'EntryContext': context
    }
```
 - looks like every entry data is under `_raw` filed so don't forget to extract it
 2. I think we can add it to context also (under `Splunk.Search` maybe as is)

File: Integrations/integration-SplunkPy.yml [hunk 5]
```diff
@@ -138,6 +138,31 @@ script:
...
+        searchquery_oneshot = demisto.args()['query']
+        if not searchquery_oneshot.startswith('search'):
+            searchquery_oneshot = 'search '+searchquery_oneshot
+        oneshotsearch_results = service.jobs.oneshot(searchquery_oneshot, **kwargs_oneshot)
+
+        reader = results.ResultsReader(oneshotsearch_results)
+        res = []
+        for item in reader:
+            if isinstance(item, results.Message):
+                demisto.results({"Type": 1, "ContentsFormat": "json", "Contents": json.dumps(result.message)})
```
Comment (asafshen) [inline Integrations/integration-SplunkPy.yml]: `json.dumps(result.message)` => `json.dumps(results.message)`

Comment (meirwah) [reply Integrations/integration-SplunkPy.yml]: results is a lib param, it's ok

**Comment by meirwah** [comment]
@asafshen plz review fixes

**Pull request merged**